Sources/SwifterSwift/UIKit/UIViewExtensions.swift

```diff
@@ -373,42 +361,20 @@ public extension UIView {
     /// SwifterSwift: Rotate view by angle on relative axis.
     ///
     /// - Parameters:
-    ///   - angle: angle to rotate view by.
-    ///   - type: type of the rotation angle.
-    ///   - animated: set true to animate rotation (default is true).
-    ///   - duration: animation duration in seconds (default is 1 second).
-    ///   - completion: optional completion handler to run with animation finishes (default is nil).
+    ///   - angle: Angle to rotate view by.
+    ///   - animated: Set `true` to animate rotation (default is `true`).
+    ///   - duration: Animation duration in seconds (default is 1 second).
+    ///   - completion: Optional completion handler to run with animation finishes (default is nil).
+    @available(watchOS 3.0, tvOS 10.0, *)
     func rotate(
-        byAngle angle: CGFloat,
-        ofType type: AngleUnit,
+        by angle: Measurement<UnitAngle>,
         animated: Bool = false,
         duration: TimeInterval = 1,
         completion: ((Bool) -> Void)? = nil) {
-        let angleWithType = (type == .degrees) ? .pi * angle / 180.0 : angle
         let aDuration = animated ? duration : 0
+        let radians = CGFloat(angle.converted(to: .radians).value)
         UIView.animate(withDuration: aDuration, delay: 0, options: .curveLinear, animations: { () -> Void in
```

Comment:
Personally I think it's really bad that this is wrapped inside an animation block. It's so unnecessary. Let whoever wants to rotate the view wrap it themselves.

Reply:
+1

Reply:
Hmmm...I didn't think about it before.

My initial intention was, the way using a enum to tell the angle value is in degrees or radians doesn't look the swift way.

I got hint from @LucianoPAlmeida that `Measurement<UnitAngle>` can be a good solution, so I just modified the parameter type and name from the old function and didn't think much about the function itself.

After you pointed it out, I reviewed the functions in `UIViewExtensions`, found that there are four functions in this kind. Two for rotating, which are merged into one in this PR, one for scaling, and one for shaking.

* `rotate(byAngle:ofType:animated:duration:completion:)`
* `rotate(toAngle:ofType:animated:duration:completion:)`
* `scale(by:animated:duration:completion:)`
* `shake(direction:duration:animationType:completion:)`

I believe the original author for these functions is trying to add extensions for simple view animations, that explains why these function signatures look like the one from `UIKit`, for example `UIScrollView.zoom(to:animated:)` or `UISwitch.setOn(_:animated:)`.

I think these functions should be re-considered the functionality they provide.

Do you think we should modify them in this PR, or we can create an issue and make some discussion there?

Reply:
I'm happy to leave this PR for just removing the old `AngleUnit` enum. However, notice that you've removed 2 functions and only added 1.

Reply:
Sorry I didn't get your point here.

The removed functions are identical in their functionality, so I think we should only keep one of them.
In the meantime, `Measurement` is used to replace the parameter in the remaining function and the argument label is also modified.

Or do you suggest, we should only remove `AngleUnit`, keeping the 2 functions in this PR, and then in another PR, deprecating the duplicated function?

I think there are two more modifications can be done to these functions. I'm also considering if they should be split into multiple PRs or put all of them in one single PR.

Reply:
Originally there was a `rotate(byAngle:` function and `rotate(toAngle:`. You've replaced the first but just deleted the second.

Reply:
Yeah, because they are doing the same thing.

Maybe we should re-organize these modifications so the history would be more clear?
Something like:

1. Deprecate `rotate(toAngle:` because duplicated functionality to `rotate(byAngle:`
2. Add new function `rotate(by:)`, which takes a `Measurement` and not wrapped in an animation block.
3. Deprecate `rotate(byAngle:`.
4. Deprecate `AngleUnit` since no function use it.

What do you think?

Reply:
Oh damn, you're right, the original `rotate(toAngle:)` was broken. I don't think it intentionally was meant to do that.
Do you know how to fix this? It's not straightforward, so if not we can just deprecate it.